presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java

```diff
@@ -79,7 +79,7 @@ public abstract class AbstractVerification<B extends QueryBundle, R extends Matc
     private static final String SNAPSHOT_DOES_NOT_EXIST = "SNAPSHOT_DOES_NOT_EXIST";
 
     private final QueryActions queryActions;
-    private final SourceQuery sourceQuery;
+    private SourceQuery sourceQuery;
```

Comment:
nit: consider keeping it final and return a new instance in `VerificationResult` (instead of `this`)

Reply:
A good catch. Also thought it's better if I can make this class immutable. However, I can't figure out how the constructor can look like, esp. since this class AbstractVerification is an abstract class. Please let me know if you have a good suggestion.

I thought I could add an abstract method to this class
```
public abstract AbstractVerification withSourceQuery(SourceQuery sourceQuery);
```
Then its subclass DataVerification overrides this method, creates and returns a DataVerification instance. However, we can't access those private fields, ex. queryActions, of the base abstract class in the subclass when calling the DataVerification constructor.

Reply:
Hi @arhimondr , I figured out a different way that doesn't need to make this sourceQuery field mutable. The original objective is to pass a signal so that the "resubmitted" verification workflow will not reuse output tables. Previously I constructed a new SourceQuery instance. A different way I figured is to check the `VerificationContext.getResubmissionCount()` as below. I figured I'm able to access VerificationContext when calling `queryRewriter.rewriteQuery()`
```
    @Override
    protected QueryObjectBundle getQueryRewrite(ClusterType clusterType)
    {
        return queryRewriter.rewriteQuery(getSourceQuery().getQuery(clusterType), getSourceQuery().getQueryConfiguration(clusterType), clusterType,
                getVerificationContext().getResubmissionCount() == 0);
    }
```